bin/i18n/utils/sync_in_base.rb

```diff
@@ -1,10 +1,15 @@
 require_relative '../i18n_script_utils'
+require_relative '../metrics'
 
 module I18n
   module Utils
     class SyncInBase
       def self.perform
-        new.send(:perform)
+        resource_class = name.match(/.*::(?<component>\w+)::SyncIn/)
```

Comment:
Since we might (and will) have a few similar resource module names, I would recommend also capturing the top-level resource name (e.g. `Dashboard::TextToSpeech`):
1. https://github.com/code-dot-org/code-dot-org/blob/staging/bin/i18n/resources/dashboard/text_to_speech/sync_out.rb#L14
2. https://github.com/code-dot-org/code-dot-org/pull/54519/files#diff-620c5364753a0a29ac54209cc6e1e108ef7358d3797ac7d25530ac9379a0aa0fR14